server/storage/types.go

```diff
@@ -4,7 +4,29 @@ import "github.com/docker/notary/tuf/data"
 
 // MetaUpdate packages up the fields required to update a TUF record
 type MetaUpdate struct {
-	Role    data.RoleName
-	Version int
-	Data    []byte
+	Role     data.RoleName
+	Version  int
+	Data     []byte
+	Channels []*Channel
```

Comment:
> for some reason GH won't let me reply to it

I think that's because I started a review, but in my "review" I replied to someone else's comment, and github posts my response both as part of my review and as part of the original comment.  :|  You might be able to respond in the original comment.

> The SQL implementation gives you what you would expect from the interfaces: when you Get from a set of channels, the file must exist in all of them, and if you Update to a set of channels, the metadata will be placed in all of them.

Ah ok, thanks for explaining.  Apologies, I had misunderstood what the intended behavior of `Get*` in the SQL implementation due to the fact that I missed the `PRIMARY KEY (tuf_file_id, channel_id)` in the channel table creation SQL, and because I just saw `Where("channel_id IN (?)", channelIDs)` in `TufFilesInChannels` without reading further.

>  think you're just seeing the deficiencies of the in-memory implementation. For testing purposes I didn't think it was necessary to try to replicate everything that sql gives you, but maybe it is worth the effort?

I think the intention of the in-memory implementation is to provide basically a verified fake for others to use, and as a reference implementation for how everything else works.  I'm ok with leaving it for now and we can address that in a future PR, so long as we document the deficiency.

I do think we should add more test cases for getting data from multiple channels, though, to be shared across the implementations of the storage interface (e.g. `server/storage/storage_test.go`, although in the in-memory implementation tests can just not call that particular function), and document this expected behavior in the interface, just because it can be easier to see what the intended behavior is if someone (like me :D) isn't super familiar with SQL joins.  I'd also be afraid of making a change in that SQL implementation code and breaking it!